spec/student_spec.rb

```diff
@@ -0,0 +1,29 @@
+require './student'
+require './classroom'
+
+describe Student do
+  context 'Testing Student class functionality' do
+    it 'Returns student details' do
+      student = Student.new(18, 'Jhon', true)
+      expect(student.age).to eq 18
+      expect(student.name).to eq 'Jhon'
+    end
+
+    it 'Return ¯(ツ)/¯ if the student plays hooky' do
+      student = Student.new(18, 'Jhon', true)
+      expect(student.play_hooky).to eq '¯(ツ)/¯'
+    end
+
+    it 'returns a hash with the student details' do
+      student = Student.new(17, 'Jhon', true)
+      id = student.instance_variable_get(:@id)
+      expect(student.to_h).to eq({
+                                   id: id,
+                                   age: 17,
+                                   name: 'Jhon',
+                                   parent_permission: true,
+                                   rentals: []
+                                 })
+    end
+  end
+end
```

Comment:
- [ ] The `Student` class has a method that sets the classroom variable in it. Kindly test this method as it is a requirement to test all methods in a class. 👍

Reply:
Hi @Gambit142,
Thanks, but which method do you? I think I have added test for all methods in student.rb

Reply:
@batoorsohail, the method that sets the student's classroom

```ruby
def classroom=(classroom)
  @classroom = classroom
  classroom.students.push(self) unless classroom.students.include?(self)
end
```

Reply:
Thanks @Gambit142  😊